src/io/dataset.cpp

```diff
@@ -608,7 +612,9 @@ void Dataset::ExtractFeaturesFromFile() {
       parser_->ParseOneLine(lines[i].c_str(), &oneline_features, &tmp_label);
       // set initial score
       if (init_score != nullptr) {
-        init_score[start_idx + i] = static_cast<float>(predict_fun_(oneline_features));
+        for (int k = 0; k < num_class_; ++k){
+            init_score[k * num_class_ + start_idx + i] = static_cast<float>(predict_fun_(oneline_features)[k]);
```

Comment:
you should call predict_fun_(oneline_features)[k] outside of loop to avoid call this repeatedly